src/VisualStudio/Core/Test/DebuggerIntelliSense/TestState.vb

```diff
@@ -179,104 +167,133 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.DebuggerIntelliSense
             MyBase.SendInvokeCompletionList(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
         End Sub
 
-        Public Overloads Sub SendCommitUniqueCompletionListItem()
-            Dim handler = DirectCast(CompletionCommandHandler, IChainedCommandHandler(Of CommitUniqueCompletionListItemCommandArgs))
-            MyBase.SendCommitUniqueCompletionListItem(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
-        End Sub
-
-        Public Overloads Sub SendSelectCompletionItemThroughPresenterSession(item As CompletionItem)
-            AssertNoAsynchronousOperationsRunning()
-            CurrentCompletionPresenterSession.SetSelectedItem(item)
-        End Sub
-
         Public Overloads Sub SendToggleCompletionMode()
-            Dim handler = DirectCast(CompletionCommandHandler, IChainedCommandHandler(Of ToggleCompletionModeCommandArgs))
+            Dim handler = DirectCast(CompletionCommandHandler, VSCommanding.ICommandHandler(Of ToggleCompletionModeCommandArgs))
             MyBase.SendToggleCompletionMode(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
         End Sub
 
-        Public Async Function AssertNoCompletionSession(Optional block As Boolean = True) As Task
-            If block Then
-                Await WaitForAsynchronousOperationsAsync()
+        Public Function HasSuggestedItem() As Boolean
+            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(TextView)
+            Assert.NotNull(session)
+            Dim computedItems = session.GetComputedItems(CancellationToken.None)
+            Return computedItems.SuggestionItem IsNot Nothing
+        End Function
+
+        Public Async Function AssertNoCompletionSession() As Task
+            Await WaitForAsynchronousOperationsAsync()
+            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(TextView)
+            If session Is Nothing Then
+                Return
+            End If
+
+            If session.IsDismissed Then
+                Return
             End If
 
-            Assert.Null(Me.CurrentCompletionPresenterSession)
+            Dim completionItems = session.GetComputedItems(CancellationToken.None)
+            ' During the computation we can explicitly dismiss the session or we can return no items.
+            ' Each of these conditions mean that there is no active completion.
+            Assert.True(session.IsDismissed OrElse completionItems.Items.Count() = 0, "AssertNoCompletionSession")
         End Function
 
         Public Async Function AssertCompletionSession() As Task
             Await WaitForAsynchronousOperationsAsync()
-            Assert.NotNull(Me.CurrentCompletionPresenterSession)
+            Dim view = TextView
+
+            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(view)
+            Assert.True(session IsNot Nothing, "AssertCompletionSession")
```

Comment:
```suggestion
            Assert.True(session IsNot Nothing, NameOf(AssertCompletionSession))
```

Reply:
Error BC36946 The implicit return variable of an Iterator or Async method cannot be accessed. 


---
In reply to: [322477414](https://github.com/dotnet/roslyn/pull/38569#discussion_r322477414) [](ancestors = 322477414)